### PR TITLE
plat-stm32mp1: prepare clock driver for generic clk framework

### PR DESCRIPTION
**Pull request by etienne-lms**
A series a change in stm32mp1 clock driver to ease integration in generic clk framework, as proposed in RFC P-R https://github.com/OP-TEE/optee_os/pull/4942.

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 1]
```diff
@@ -387,6 +387,13 @@ static const struct stm32mp1_clk_gate stm32mp1_clk_gate[] = {
 };
 DECLARE_KEEP_PAGER(stm32mp1_clk_gate);
 
+const uint8_t stm32mp1_clk_on[] = {
+	CK_HSE, CK_CSI, CK_LSI, CK_LSE, CK_HSI, CK_HSE_DIV2,
+	PLL1_Q, PLL1_R, PLL2_P, PLL2_Q, PLL2_R, PLL3_P, PLL3_Q, PLL3_R, PLL3_R,
+	CK_AXI, CK_MPU, CK_MCU,
+};
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:394]: `PLL1_P` seems to be missing

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 2]
```diff
@@ -387,6 +387,13 @@ static const struct stm32mp1_clk_gate stm32mp1_clk_gate[] = {
 };
 DECLARE_KEEP_PAGER(stm32mp1_clk_gate);
 
+const uint8_t stm32mp1_clk_on[] = {
+	CK_HSE, CK_CSI, CK_LSI, CK_LSE, CK_HSI, CK_HSE_DIV2,
+	PLL1_Q, PLL1_R, PLL2_P, PLL2_Q, PLL2_R, PLL3_P, PLL3_Q, PLL3_R, PLL3_R,
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:392]: `PLL3_R` appears twaice

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 3]
```diff
@@ -940,32 +947,13 @@ static bool __clk_is_enabled(struct stm32mp1_clk_gate const *gate)
...
-		return true;
-	default:
-		return false;
-	}
+	for (n = 0; n < ARRAY_SIZE(stm32mp1_clk_on); n++)
+		if (stm32mp1_clk_on[n] == id)
+			return true;
+
+	return false;
 }
```
Comment (jforissier) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:952]: Why `DECLARE_KEEP_PAGER(stm32mp1_clk_on);` and not `DECLARE_KEEP_PAGER(clock_is_always_on);`?
Could `stm32mp1_clk_on` be local to `clock_is_always_on`?

**Review by jforissier — COMMENTED** [review]
Some comment on "plat-stm32mp1: clk: refine always on clocks ".

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c (hunk 2 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:392]: thanks

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:394]: thanks!

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c (hunk 3 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c:952]: Right, i was to hurry on this. No need for `DECLARE_KEEP_PAGER(stm32mp1_clk_on)`, same for   `clock_is_always_on()`: they will already be embedded into unpaged sections if needed using the standard dependency chains.

**Comment by jforissier** [comment]
s/attriibure/attribute/, with that:
`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Comment by etienne-lms** [comment]
@jforissier, could you merge this P-R?

**Comment by jforissier** [comment]
@etienne-lms sure, thanks for reminding me :)

**Pull request merged**